proposals/selfAssemblingModelHierarchy/README.md

```diff
@@ -0,0 +1,342 @@
+
+```
+# `kind` has to be read on the prims where it is explicitly
+# specified plus the `/root/assembly/group` prim.
+# It is not read on the descendants of the `component` models.
+def "root" (kind = "group") {
+    def "assembly" (kind = "assembly") {
+        def "group" {
+            def "component_1" (kind = "component") {
+                def "gprim_1" {}
+                def "gprim_2" {}
+            }
+            def "component_2" (kind = "component") {
+                def "gprim_1" {}
+                def "gprim_2" {}
+            }
+        }
+    }
+}
+```
+Any prim that is a child of a `group` (or `assembly`) prim not explicitly specified will pay the cost of the pseudo root prim flag check. In this case, that's only `/root/assembly/group`.
+
+### Performantly Handling an Incomplete Hierarchy
+The fundamental challenge of this proposal is determining how to performantly handle an incomplete hierarchy.
+
+```
+# An incomplete model hierarchy-- The "East River Conundrum"
+def Xform "NewYorkCity" (kind = "assembly") {
+    def Scope "Water" (kind = "group") {
+        def Mesh "EastRiver" {}
+        def Material "EastRiverMaterial" { ... }
+    }
+}
+```
+
+The "EastRiver" is geometry just inlined as a mesh into the "NewYorkCity" assembly with a material. This "incomplete" hierarchy is not incorrect.  It just means that there are leaf prims that are not `component` descendants.
+
+It's worth noting that in the current implementation without propagation, `kind` will be checked on the `EastRiver` and `EastRiverMaterial` scopes. **If we can signal that propagation should end on these scopes, then we can provide a path for self assembling model hierarchies without additional `kind` reads.**
+
+#### Variant #1: Introduce a new `auto` kind which enables propagation
+`auto` becomes `kind`'s new fallback value. The "EastRiver" and other members of the incomplete hierarchy could be explicitly tagged with `kind=""` to preserve current behavior. Propagation for `group` or `assembly` kinds is `group`.
+
+#### Variant #2: Use authored state to control propagation
+Only propagate when `kind` is unauthored. This is effectively the same as `auto` in that "EastRiver" can be tagged with `kind=""` to suppress propagation.
+
+In user interfaces, it may be hard to disambiguate between unauthored (and propagating) and authored (not propagating) `kind` state. There would also not be way to author a value to re-enable propagation, but this may be fine.
+
+#### Variant #3: Use `subcomponent` (or some other kind) to prune traversal
+Despite its name, there are no rules that `subcomponent` prims are descendants of `component` prims. One could use that or some other kind to meaningfully stop traversal. `kind=none` (or `kind=terminal`) has been another suggestion. Even if variant #1 or #2 were adopted, this would still be a valid way to prune traversal, it just wouldn't be the only (or preferred) way.
```

Comment:
Given that we are only ever proposing to propagate when the relevant root prim is explicitly tagged as a model container, why not just say that `component` is the preferred terminator?  What's the point in declaring a model-containing root prim if you are not going to add any models into it?  And if it's a case of "some assets/prims have proper component tagging, but others don't, it seems the pattern is pretty clear and someone (or some software) just forgot to properly tag some asset, then let's address the traversal-pruning problem by adding components.  As you argue below, even adding `subcomponent` tagging to prune traversals _could_ change the contents of some collection path expressions, so if we're going to "break" things, let's break them in the most organized way ?